common/data_refinery_common/models/models.py

```diff
@@ -407,11 +412,24 @@ def __str__(self):
     # This matters, for instance salmon 0.9.0 indexes don't work with 0.10.0
     salmon_version = models.CharField(max_length=255)
 
+    # S3 Key
+    s3_key = models.CharField(max_length=255, default="")
```

Comment:
You can't store `s3_key` without storing `s3_bucket` along with it as well, since both are needed to actually retrieve the value. It's okay to get the bucket location from an envvar at runtime when _uploading_, but it must be retrieved from a datastore when the value is being _retrieved_, else they could desync and we could lose the value.